File: community/cypher/cypher-compiler-2.0/src/main/scala/org/neo4j/cypher/internal/compiler/v2_0/blackbuck/impl/operator/FusingOperator.java
package org.neo4j.cypher.internal.compiler.v2_0.blackbuck.impl.operator;

import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.sched.Operator;
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.slot.Cursor;

public abstract class FusingOperator<C extends Cursor<C>> extends ChainingOperator<C> {
    private final Operator<C> lhs;
    private final Operator<C> rhs;

    public FusingOperator( Object key, Operator<C> destination, Operator<C> lhs, Operator<C> rhs )
    {
        super( key, destination );
        this.lhs = lhs;
        this.rhs = rhs;
    }

    public Operator<C> lhs() {
        return lhs;
    }

    public Operator<C> rhs() {
        return rhs;
    }
}

File: community/cypher/cypher-compiler-2.0/src/main/scala/org/neo4j/cypher/internal/compiler/v2_0/blackbuck/impl/operator/FusingOperator.scala
package org.neo4j.cypher.internal.compiler.v2_0.blackbuck.impl.operator

import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.slot.Pos
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.rows.{RowPool, Cursor}
import org.neo4j.cypher.internal.compiler.v2_0.blackbuck.api.sched.{Scheduler, Operator}

abstract class FusingOperator[P <: Pos[P], C <: Cursor[P, C], E](key: AnyRef, dest: Operator[P, C, E],
                                                                 val lhs: Operator[P, C, E],
                                                                 val rhs: Operator[P, C, E])
  extends ChainingOperator[P, C, E](key, dest) {
}
